### PR TITLE
chore: remove deprecated minishift/minikube commands

### LINKED ISSUE #4088: Remove Minishift/Minikube cluster options
### Summary 

Running Jenkins-X on Minishift/Minikube does not seem to be terribly useful given that these single node clusters won't be reachable from the outside, meaning that webhooks et all won't work.

There are also resource constraints. A full Jenkins-X install with all its components just takes too many resources for such a single node cluster.

Last but not least, Minishift is now in maintenance mode. It won't work with OpenShift 4 anymore.

### PR DESCRIPTION
**Pull request by mrageh**
We've deprecated the minishift command a while back, this commit removes it from jx.

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #4088

### CODE REVIEW

**Comment by mrageh** [comment]
/hold 

Let's not merge this until we figure out how we're going to version all the deprecated commands

File: README.md [hunk 1]
```diff
@@ -252,92 +248,21 @@ For example to add the [Gitea Git server](https://gitea.io/en-US/) to your Jenki
...
 * add Gitea as a Git server (via the `jx create git server gitea` command).
 * create a new user in Gitea (via the `jx create git user -n gitea` command).
 * create a new Git API token in Gitea (via the `jx create git token -n gitea -p password username` command).
-     
+
 ## Troubleshooting
 
 We have tried to collate common issues here with work arounds. If your issue isn't listed here please [let us know](https://github.com/jenkins-x/jx/issues/new).
- 
-### Cannot create cluster Minikube
```
Comment (hferentschik) [inline README.md]: 👍

Comment (hferentschik) [reply README.md]: I think there is another issue just around this README update. https://github.com/jenkins-x/jx/issues/6799

Maybe while on it, we should add some other information into the README. @dennisnewel

File: pkg/cmd/create/create_cluster.go [hunk 2]
```diff
@@ -116,7 +112,6 @@ func NewCmdCreateCluster(commonOpts *opts.CommonOptions) *cobra.Command {
 	cmd.AddCommand(NewCmdCreateClusterEKS(commonOpts))
 	cmd.AddCommand(NewCmdCreateClusterGKE(commonOpts))
 	cmd.AddCommand(NewCmdCreateClusterMinikube(commonOpts))
```
Comment (hferentschik) [inline pkg/cmd/create/create_cluster.go]: That probably needs removing as well

File: pkg/cmd/create/create_etc_hosts.go [hunk 3]
```diff
@@ -75,14 +75,6 @@ func (o *CreateEtcHostsOptions) Run() error {
 	if name == "" {
 		return util.MissingOption(name)
 	}
-	if o.IP == "" {
```
Comment (hferentschik) [inline pkg/cmd/create/create_etc_hosts.go]: I think we safely can remove the whole command

File: README.md (hunk 1 above)
Comment (dennisnewel) [reply README.md]: are you thinking about adding a deprecation/removal section or something? e.g. "We removed Minicube support in version 2.0.xxx"?

Comment (hferentschik) [reply README.md]: @dennisnewel, I was more thinking of whether we should link to a more general FAQ. I would have thought we have a FAQ section in the online docs, but that does not seem to be the case. I don't think the README is a good location for this type of troubleshooting question. If there is no FAQ in the docs, we should create one there.

Regarding Minishift/Minikube, I don't think any deprecation section is needed. Definitely not in the README.

File: pkg/cmd/create/create_etc_hosts.go (hunk 3 above)
Comment (hferentschik) [reply pkg/cmd/create/create_etc_hosts.go]: @mrageh, I meant we can remove the whole create_etc_host command.

**Comment by mrageh** [comment]
Closing in favour of #6947

**Pull request closed without merging**